packages/@aws-cdk/aws-stepfunctions-tasks/lib/invoke-function.ts

```diff
@@ -2,26 +2,53 @@ import iam = require('@aws-cdk/aws-iam');
 import lambda = require('@aws-cdk/aws-lambda');
 import sfn = require('@aws-cdk/aws-stepfunctions');
 
+/**
+ * Properties for InvokeFunction
+ */
+export interface InvokeFunctionProps {
+  /**
+   * The JSON that you want to provide to your Lambda function as input.
+   */
+  readonly payload?: { [key: string]: string };
+
+  /**
+   * Whether to pause the workflow until a task token is returned
+   *
+   * @default false
+   */
+  readonly waitForTaskToken?: boolean;
+}
+
 /**
  * A StepFunctions Task to invoke a Lambda function.
  *
  * A Function can be used directly as a Resource, but this class mirrors
  * integration with other AWS services via a specific class instance.
  */
 export class InvokeFunction implements sfn.IStepFunctionsTask {
-  constructor(private readonly lambdaFunction: lambda.IFunction) {
+
+  private readonly waitForTaskToken: boolean;
+
+  constructor(private readonly lambdaFunction: lambda.IFunction, private readonly props: InvokeFunctionProps) {
+    this.waitForTaskToken = props.waitForTaskToken === true;
   }
 
   public bind(_task: sfn.Task): sfn.StepFunctionsTaskProperties {
     return {
-      resourceArn: this.lambdaFunction.functionArn,
+      resourceArn: this.waitForTaskToken
+        ? 'arn:aws:states:::lambda:invoke.waitForTaskToken'
+        : this.lambdaFunction.functionArn,
       policyStatements: [new iam.PolicyStatement()
         .addResource(this.lambdaFunction.functionArn)
         .addActions("lambda:InvokeFunction")
       ],
       metricPrefixSingular: 'LambdaFunction',
       metricPrefixPlural: 'LambdaFunctions',
       metricDimensions: { LambdaFunctionArn: this.lambdaFunction.functionArn },
+      parameters: {
+        FunctionName: this.lambdaFunction.functionName,
+        ...this.props.payload && { Payload: this.props.payload },
```

Comment:
I don't think `Parameters.Payload` will work in case of a task invocation. Will it?

Reply:
I would also like an additional validation that if `waitForTaskToken` is used, the magic token substitution field is used somewhere in the payload. If not, that should be an error. For all classes that implement this.

Reply:
So, if `props.waitForTaskToken` is true, I have to validate `props.payload` in order to check if something like `"token.$":"$$.Task.Token"` is present. Did I get it right?

Reply:
Actually I would like you to make a new function on `JsonPath` (something like `JsonPath.taskToken()`, similar to the other functions), and make sure one of the fields in the `payload` structure contains the value returned. Be aware that the payload could be a deep structure, so you will need to recurse into it.

Reply:
I think I will do the start of this. There is some work I want done in the SFN module anyway.

Reply:
Okay, I await your updates

Reply:
What you're going to need is in https://github.com/awslabs/aws-cdk/pull/2706

Reply:
That other PR has been merged. You may continue :).

Please also have a look at what @wqzoww has written.

Reply:
Hello @rix0rrr only one thing: may I ask you how to use `FieldUtils.containsTaskToken` to validate the payload in order to check if the token is present?

Reply:
I would expect you to call it on the payload structure in the constructor, and throw an error if the function returns `false`.

Reply:
Ok, i've done it in the latest commit. But tests (unit and integ) are failing now, could you take a look please?